src/system_metrics/mongoose_system_metrics_collector.erl

```diff
@@ -86,3 +92,80 @@ get_number_of_custom_modules() ->
     MetricsModuleSet = sets:from_list(MetricsModule),
     CountCustomMods= sets:size(sets:subtract(GenModsSet, MetricsModuleSet)),
     #{report_name => custom_modules, key => count, value => CountCustomMods}.
+
+get_uptime() ->
+    {Uptime, _} = statistics(wall_clock),
+    UptimeSeconds = Uptime div 1000,
+    {D, {H, M, S}} = calendar:seconds_to_daystime(UptimeSeconds),
+    Formatted = io_lib:format("~4..0B-~2..0B:~2..0B:~2..0B", [D,H,M,S]),
+    [#{report_name => cluster, key => uptime, value => Formatted}].
+
+get_cluster_size() ->
+    NodesNo = length(nodes()) + 1,
+    [#{report_name => cluster, key => number_of_nodes, value => NodesNo}].
+
+get_version() ->
+    case lists:keyfind(mongooseim, 1, application:which_applications()) of
+        false -> Version = none;
+        {_, _, Ver} -> Version = Ver
+    end,
+    #{report_name => cluster, key => mim_version, value => Version}.
+
+get_components() ->
+    Domains = ejabberd_router:dirty_get_all_domains(),
+    Hosts = ejabberd_config:get_global_option(hosts),
+    Components = lists:flatten(
+                    lists:map(
+                        fun(Host) ->
+                            check_components(Host, Domains)
+                        end, Hosts)),
+    LenComponents = length(Components),
+    #{report_name => cluster, key => number_of_components, value => LenComponents}.
+
+check_components(Host, Domains) ->
+    lists:flatten(
+        lists:map(
+            fun(Domain) ->
+                ejabberd_router:lookup_component(Domain, Host)
+            end, Domains)).
+
+get_api() ->
+    ModulesOptions = lists:flatten(get_service_option(ejabberd_cowboy, modules)),
+    ApiList = lists:usort(lists:map(
+                fun(Module)->
+                    element(3, Module)
```

Comment:
Could you please add a short explanation why we use the `element` function in stead of patter matching on an element in a tuple?